components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -1141,8 +1142,14 @@ public static void main(String[] args) throws Exception {
 
       while (scanner.hasNext()) {
         newArgs[idx] = scanner.nextLine();
-	System.out.println("====% " + newArgs[idx]);
-        if (!new ImageInfo().testRead(newArgs)) System.exit(1);
+        System.out.println("====% " + newArgs[idx]);
+        try {
+        	new ImageInfo().testRead(newArgs);
+        }
+        catch (UnknownFormatException e) {
+			LOGGER.warn("Unknown file format: " + newArgs[idx]);
+        	continue;
```

Comment:
What should the return code be if any of the processed files fails? With the current implementation it will be zero

Reply:
I definitely see the value of returning a non-zero return code if any input file fails. This would give an indication to the called that the output of the utility should be reviewed for exceptions.